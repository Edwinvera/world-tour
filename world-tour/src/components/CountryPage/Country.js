import React from 'react'
import {useParams, Link} from 'react-router-dom'
import CountUp from 'react-countup'
import styled from 'styled-components'

const Main = styled.main`
  display: flex;
  align-items: center;
  flex-direction: column;
  text-align: left;
  margin: 250px 0 0 0;
  padding-bottom: 80px
`
const Summary = styled.section`
  width: 500px;
  background-color: rgba(20,20,20,.1);
  box-shadow: 5px 10px 20px rgb(20,20,20);
  padding: 40px;
  margin-bottom: 50px
`
const Header = styled.h1`
  margin-bottom: 5px;
  font-family: 'Nunito Sans', sans-serif;
  font-size: 36px;
  letter-spacing: 5px;
`
const NativeName = styled.i`
  font-size: 12px;
  letter-spacing: 5px;
`
const Divider = styled.hr`
  width: 500px;
  margin: 18px 0px
`
const Table = styled.table`
  width: 500px
`
const Label = styled.td`
  font-family: 'Nunito Sans', sans-serif;
  font-size: 18px;
  line-height: 2
`
const Details = styled.td`
  font-family: sans-serif;
  font-size: 18px;
  line-height: 2;
  text-align: right
`
const Flag = styled.img`
  width: 600px;
  border: grey solid 2px
`
const ReturnBtn = styled.button`
  font-size: 18px;
  border: solid 1px rgba(20,20,20,.2);
  border-radius: 15px;
  background-color: rgba(60,60,155,.1);
  cursor: pointer;
`
const RegionLink = styled(Link)`
`

function Country({regions}) {

  
  const {countryName} = useParams()
  let countryData = regions.filter(r => r.name === countryName)
  let country = countryData[0]
  console.log(country)


  return (
    <div>
      <Main>
        <Summary>
          <RegionLink to={`/region/${country.region}`}>
            <ReturnBtn>{`${`â¬…`} ${country.region}`}</ReturnBtn>
          </RegionLink>
          <Header>{countryName}</Header>
          <NativeName>( {country.nativeName} )</NativeName>
          <Divider/>
          <Table>
            <tbody>
              <tr>
                <Label>{`Population in ${countryName}:`}</Label>
                <Details>
                  <CountUp
                    end={country.population}
                    separator=','
                  />
                </Details>
              </tr>
              <tr>
                <Label>Area of Land (mi<sup>2</sup>):</Label>
                <Details>
                  <CountUp
                    end={country.area}
                    separator=','
                  />
                </Details>
              </tr>
              <tr>
                <Label>Capital:</Label>
                <Details>{country.capital}</Details>
              </tr>
              <tr>
                <Label>Currency (Symbol):</Label>
                <Details>{country.currencies[0].code} ({country.currencies[0].symbol})</Details>
              </tr>
              <tr>
                <Label>Region:</Label>
                <Details>{country.region}</Details>
              </tr>
              <tr>
                <Label>Sub Region:</Label>
                <Details>{country.subregion}</Details>
              </tr>
            </tbody>
          </Table>
          <Divider />
        </Summary>
        <a
          href={`https://www.google.com/maps/place/${countryName}/@${country.latlng[0]},${country.latlng[1]},6z/`}
          target="_blank"
          rel="noopener noreferrer"
        >
          <Flag src={country.flag} alt={countryName}/>
        </a>  
      </Main>
    </div>
  )
}

export default Country
