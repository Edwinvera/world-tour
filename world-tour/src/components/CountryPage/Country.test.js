import renderer from 'react-test-renderer'
import 'jest-styled-components'
import React from 'react'
import { Flag, Main, Label} from './Country'



test('Country_Flag component renders', () => {
  const FlagComponent = renderer.create(<Flag/>).toJSON()
  expect(FlagComponent).toMatchSnapshot()
})

test('Country_Main component renders', () => {
  const MainComponent = renderer.create(<Main/>).toJSON()
  expect(MainComponent).toMatchSnapshot()
})

test('Country_Label component renders', () => {
  const LabelComponent = renderer.create(<Label/>).toJSON()
  expect(LabelComponent).toMatchSnapshot()
})
