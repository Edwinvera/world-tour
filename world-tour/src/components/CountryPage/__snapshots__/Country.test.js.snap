// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country_Flag component renders 1`] = `
.c0 {
  width: 500px;
  border: grey solid 2px;
}

.c0:hover {
  box-shadow: 5px 10px 40px grey;
  opacity: .9;
}

@media (max-width:1050px) {
  .c0 {
    width: 800px;
  }
}

@media (max-width:900px) {
  .c0 {
    width: 500px;
  }
}

@media (max-width:697px) {
  .c0 {
    width: 300px;
  }
}

<img
  className="c0"
/>
`;

exports[`Country_Label component renders 1`] = `
.c0 {
  font-family: 'Nunito Sans',sans-serif;
  font-size: 18px;
  line-height: 2;
}

@media (max-width:1050px) {
  .c0 {
    font-size: 24px;
    text-align: left;
  }
}

@media (max-width:697px) {
  .c0 {
    font-size: 18px;
  }
}

<td
  className="c0"
/>
`;

exports[`Country_Main component renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  margin: 250px 0 0 0;
  padding-bottom: 80px;
}

<main
  className="c0"
/>
`;
